### PR TITLE
Added doco for dataloaders

### PR DESCRIPTION
**Pull request by bbakerman**

### CODE REVIEW

**Comment by bbakerman** [comment]
Once https://github.com/graphql-java/graphql-java/pull/1263 merges - we can merge this doco update

File: content/documentation/master/batching.md [hunk 1]
```diff
@@ -272,14 +318,77 @@ The following is how you can still have asynchronous code, by placing it into th
...
+                Object source = environment.getSource();
+                //
+                // you can pass per load call contexts
+                //
+                return characterDataLoader.load(argId, source);
+            }
+        };
+        
+{{< / highlight >}}
+
```
Comment (felipe-gdr) [inline content/documentation/master/batching.md]: Should these changes be copied over to other documentation directories (v10 and v9)?

Comment (bbakerman) [reply content/documentation/master/batching.md]: no - master only

File: content/documentation/master/batching.md [hunk 2]
```diff
@@ -124,24 +126,22 @@ Here is how you might put this in place:
...
-        //
-        // DataLoaderRegistry is a place to register all data loaders in that needs to be dispatched together
-        // in this case there is 1 but you can have many
-        //
-        DataLoaderRegistry registry = new DataLoaderRegistry();
-        registry.register("character", characterDataLoader);
 
         //
-        // this instrumentation implementation will dispatch all the dataloaders
+        // this instrumentation implementation will dispatched all the data loaders
```
[the remark this answers is not in the document]
Comment (bbakerman) [reply content/documentation/master/batching.md]: updated

File: content/documentation/master/batching.md [hunk 3]
```diff
@@ -94,7 +94,10 @@ Here is how you might put this in place:
 
 {{< highlight java "linenos=table" >}}
 
+        //
         // a batch loader function that will be called with N or more keys for batch loading
+        // This can be a singleton object since its stateless
```
[the remark this answers is not in the document]
Comment (bbakerman) [reply content/documentation/master/batching.md]: updated

**Pull request merged**